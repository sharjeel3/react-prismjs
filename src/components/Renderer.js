import React from 'react';
import PropTypes from 'prop-types';
import { RenderTokens } from './RenderTokens';
require(`prismjs/themes/prism-okaidia.css`);

export const Renderer = ({ tokens, language }) => {
  return (
    <pre className={`language-${language}`}>
      <code className={`language-${language}`}>
        <RenderTokens tokens={tokens} />
      </code>
    </pre>
  );
};

Renderer.propTypes = {
  tokens: PropTypes.array.isRequired,
  language: PropTypes.string.isRequired
};
