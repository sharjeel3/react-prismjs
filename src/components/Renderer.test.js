import React from 'react';
import testRenderer from 'react-test-renderer';
import { Renderer } from './Renderer';

const tokens = [
  {
    type: 'keyword',
    content: 'const',
    length: 5,
    greedy: false
  },
  ' message ',
  {
    type: 'operator',
    content: '=',
    length: 1,
    greedy: false
  },
  {
    type: 'string',
    content: '"test"',
    length: 6,
    greedy: true
  },
  {
    type: 'punctuation',
    content: ';',
    length: 1,
    greedy: false
  },
  '\n\n  ',
  {
    type: 'keyword',
    content: 'const',
    length: 5,
    greedy: false
  },
  {
    type: 'function-variable',
    content: 'sayIt',
    alias: 'function',
    length: 5,
    greedy: false
  },
  {
    type: 'operator',
    content: '=',
    length: 1,
    greedy: false
  },
  ' ',
  {
    type: 'punctuation',
    content: '(',
    length: 1,
    greedy: false
  },
  {
    type: 'punctuation',
    content: ')',
    length: 1,
    greedy: false
  },
  ' ',
  {
    type: 'operator',
    content: '=>',
    length: 2,
    greedy: false
  },
  ' ',
  {
    type: 'punctuation',
    content: '{',
    length: 1,
    greedy: false
  },
  '\n    ',
  {
    type: 'keyword',
    content: 'return',
    length: 6,
    greedy: false
  },
  ' message',
  {
    type: 'punctuation',
    content: ';',
    length: 1,
    greedy: false
  },
  '\n  ',
  {
    type: 'punctuation',
    content: '}',
    length: 1,
    greedy: false
  },
  {
    type: 'punctuation',
    content: ';',
    length: 1,
    greedy: false
  }
];

describe('Renderer', () => {
  it('matches snapshot', () => {
    const tree = testRenderer.create(<Renderer language="javascript" tokens={tokens} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
