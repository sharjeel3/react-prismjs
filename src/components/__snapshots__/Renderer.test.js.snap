// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer matches snapshot 1`] = `
<pre
  className="language-javascript"
>
  <code
    className="language-javascript"
  >
    <span
      className="token keyword"
    >
      const
    </span>
     message 
    <span
      className="token operator"
    >
      =
    </span>
    <span
      className="token string"
    >
      "test"
    </span>
    <span
      className="token punctuation"
    >
      ;
    </span>
    

  
    <span
      className="token keyword"
    >
      const
    </span>
    <span
      className="token function-variable"
    >
      sayIt
    </span>
    <span
      className="token operator"
    >
      =
    </span>
     
    <span
      className="token punctuation"
    >
      (
    </span>
    <span
      className="token punctuation"
    >
      )
    </span>
     
    <span
      className="token operator"
    >
      =&gt;
    </span>
     
    <span
      className="token punctuation"
    >
      {
    </span>
    
    
    <span
      className="token keyword"
    >
      return
    </span>
     message
    <span
      className="token punctuation"
    >
      ;
    </span>
    
  
    <span
      className="token punctuation"
    >
      }
    </span>
    <span
      className="token punctuation"
    >
      ;
    </span>
  </code>
</pre>
`;
